### DISCUSSION TITLE
Multi monitor issue

### DISCUSSION THREAD

**Discussion by abrahamesc** [i3/i3#5874] [Q&A]
I currently have 3 monitor setup. All 3 monitors are connected to a docking station. Docking station is then connected via thunderbolt to Asus M16. 

My main issue is that I cannot re-arrange the monitors via xrandr the way I want them to or turn off the laptop's internal display. The output of xrandr also has me confused. It shows that only e-DP1 and DP-1-3 are connected. However all 3 external monitors display a workspace. Here's the xrandr output:

 ```
Screen 0: minimum 320 x 200, current 8320 x 1600, maximum 16384 x 16384
eDP-1 connected primary 2560x1600+0+0 (normal left inverted right x axis y axis) 345mm x 215mm
   2560x1600    240.00*+  60.00 +  59.99    59.97
   2560x1440     59.99    59.99    59.96    59.95
   2048x1536     85.00    75.00    60.00
   1920x1440     85.00    75.00    60.00
   1856x1392     75.00    60.01
   1792x1344     75.00    60.01
   2048x1152     59.99    59.98    59.90    59.91
   1920x1200     59.88    59.95
   1920x1080     60.01    59.97    59.96    59.93
   1600x1200     85.00    75.00    70.00    65.00    60.00
   1680x1050     59.95    59.88
   1400x1050     74.76    59.98
   1600x900      59.99    59.94    59.95    59.82
   1280x1024     85.02    75.02    60.02
   1400x900      59.96    59.88
   1280x960      85.00    60.00
   1440x810      60.00    59.97
   1368x768      59.88    59.85
   1280x800      59.99    59.97    59.81    59.91
   1152x864      75.00
   1280x720      60.00    59.99    59.86    59.74
   1024x768      85.00    75.05    60.04    85.00    75.03    70.07    60.00
   1024x768i     86.96
   960x720       85.00    75.00    60.00
   928x696       75.00    60.05
   896x672       75.05    60.01
   1024x576      59.95    59.96    59.90    59.82
   960x600       59.93    60.00
   832x624       74.55
   960x540       59.96    59.99    59.63    59.82
   800x600       85.00    75.00    70.00    65.00    60.00    85.14    72.19    75.00    60.32    56.25
   840x525       60.01    59.88
   864x486       59.92    59.57
   700x525       74.76    59.98
   800x450       59.95    59.82
   640x512       85.02    75.02    60.02
   700x450       59.96    59.88
   640x480       85.09    60.00    85.01    72.81    75.00    59.94
   720x405       59.51    58.99
   720x400       85.04
   684x384       59.88    59.85
   640x400       59.88    59.98    85.08
   576x432       75.00
   640x360       59.86    59.83    59.84    59.32
   640x350       85.08
   512x384       85.00    75.03    70.07    60.00
   512x384i      87.06
   512x288       60.00    59.92
   416x312       74.66
   480x270       59.63    59.82
   400x300       85.27    72.19    75.12    60.32    56.34
   432x243       59.92    59.57
   320x240       85.18    72.81    75.00    60.05
   360x202       59.51    59.13
   360x200       85.04
   320x200       85.27
   320x180       59.84    59.32
   320x175       85.27
DP-1 disconnected (normal left inverted right x axis y axis)
DP-2 disconnected (normal left inverted right x axis y axis)
DP-1-1 disconnected (normal left inverted right x axis y axis)
DP-1-2 disconnected (normal left inverted right x axis y axis)
DP-1-3 connected 1920x1080+6400+0 (normal left inverted right x axis y axis) 598mm x 336mm
   1920x1080     60.00*+  74.99    50.00    59.94
   1680x1050     59.88
   1600x900      60.00
   1280x1024     75.02    60.02
   1440x900      59.90
   1280x960      60.00
   1366x768      59.79
   1152x864      75.00
   1280x768      59.87
   1280x720      60.00    50.00    59.94
   1024x768      75.03    60.00
   800x600       75.00    60.32
   720x576       50.00
   720x480       60.00    59.94
   640x480       75.00    60.00    59.94
   720x400       70.08
DP-1-0 disconnected (normal left inverted right x axis y axis)
HDMI-1-0 disconnected (normal left inverted right x axis y axis)
```

It's like x is aggregating all 2 ext monitors + the internal monitor into one, since only one external monitor shows up as DP-1-3. Then we have the output of xrandr --listactivemonitors which displays what I would expect xrandr to:

```
Monitors: 4
 0: +*eDP-1 2560/345x1600/215+0+0  eDP-1
 1: +DP-1-1 1920/598x1080/336+2560+0  DP-1-1
 2: +DP-1-2 1920/598x1080/336+4480+0  DP-1-2
 3: +DP-1-3 1920/598x1080/336+6400+0  DP-1-3
```

If I try to for example: xrandr --output DP-1-3 --right-of DP-1-1 --auto, the internal monitor will start mirroring DP-1-3 in a portion of the screen. 

If I login with GNOME, xrandr correctly displays each monitor and I am able to disable the internal display.

**Comment by abrahamesc** [comment]
Here's the output of xrandr from GNOME:

Screen 0: minimum 16 x 16, current 5760 x 1080, maximum 32767 x 32767
DP-3 connected primary 1920x1080+1920+0 (normal left inverted right x axis y axis) 600mm x 340mm
   1920x1080     59.96*+
   1440x1080     59.99
   1400x1050     59.98
   1280x1024     59.89
   1280x960      59.94
   1152x864      59.96
   1024x768      59.92
   800x600       59.86
   640x480       59.38
   320x240       59.52
   1680x1050     59.95
   1440x900      59.89
   1280x800      59.81
   1152x720      59.97
   960x600       59.63
   928x580       59.88
   800x500       59.50
   768x480       59.90
   720x480       59.71
   640x400       59.95
   320x200       58.96
   1600x900      59.95
   1368x768      59.88
   1280x720      59.86
   1024x576      59.90
   864x486       59.92
   720x400       59.55
   640x350       59.77
DP-4 connected 1920x1080+0+0 (normal left inverted right x axis y axis) 600mm x 340mm
   1920x1080     59.96*+
   1440x1080     59.99
   1400x1050     59.98
   1280x1024     59.89
   1280x960      59.94
   1152x864      59.96
   1024x768      59.92
   800x600       59.86
   640x480       59.38
   320x240       59.52
   1680x1050     59.95
   1440x900      59.89
   1280x800      59.81
   1152x720      59.97
   960x600       59.63
   928x580       59.88
   800x500       59.50
   768x480       59.90
   720x480       59.71
   640x400       59.95
   320x200       58.96
   1600x900      59.95
   1368x768      59.88
   1280x720      59.86
   1024x576      59.90
   864x486       59.92
   720x400       59.55
   640x350       59.77
DP-5 connected 1920x1080+3840+0 (normal left inverted right x axis y axis) 600mm x 340mm
   1920x1080     59.96*+
   1440x1080     59.99
   1400x1050     59.98
   1280x1024     59.89
   1280x960      59.94
   1152x864      59.96
   1024x768      59.92
   800x600       59.86
   640x480       59.38
   320x240       59.52
   1680x1050     59.95
   1440x900      59.89
   1280x800      59.81
   1152x720      59.97
   960x600       59.63
   928x580       59.88
   800x500       59.50
   768x480       59.90
   720x480       59.71
   640x400       59.95
   320x200       58.96
   1600x900      59.95
   1368x768      59.88
   1280x720      59.86
   1024x576      59.90
   864x486       59.92
   720x400       59.55
   640x350       59.77

You can see that here we have 3 different monitors connected 

Any suggestions?

**Reply by okraits** [reply]
Is it reproducable that xrandr only shows 2 connected outputs? You could try to configure your outputs with arandr.

xrandr configuration should be independent of the window manager - i3, Gnome's window manager or anything else. The window manager just uses the output(s) as configured.

**Reply by abrahamesc** [reply]
Yes, it happens every time. 

Arandr shows the same as xrandr
![2024-02-02_14-25](https://github.com/i3/i3/assets/56666534/f8e04665-4094-495b-aac8-6ce4c79bec07)
